agimus_controller/agimus_controller/trajectory.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+from collections import deque
+from dataclasses import dataclass
+from pinocchio import SE3, Force
+
+
+@dataclass
+class TrajectoryPoint:
+    """Trajectory point aiming at being a reference for the MPC."""
+
+    time_ns: int
+    robot_configuration: np.ndarray
+    robot_velocity: np.ndarray
+    robot_acceleration: np.ndarray
+    robot_effort: np.ndarray
+    forces: dict[Force]  # Dictionary of pinocchio.Force
+    end_effector_poses: dict[SE3]  # Dictionary of pinocchio.SE3
+
+
+@dataclass
+class TrajectoryPointWeights:
+    """Trajectory point weights aiming at being set in the MPC costs."""
+
+    w_robot_configuration: np.ndarray
+    w_robot_velocity: np.ndarray
```

Comment:
in crocoddyl there's no cost for only q or qdot so I guess we only have this instead :
```suggestion
    w_robot_state: np.ndarray
```

Reply:
the weights are consistent with the trajectory point entries. I would say we either
1) use `state` instead of `configuration` and `velocity`
or 2) keep them separate and merge in ocp to state

Also, I remember @MaximilienNaveau  mentioning that state is not always [config, velocity], so option 1) might be misleading in the future

Reply:
for the second option how would we merge it in ocp to state ? the state will indeed not always be [config, velocity], but maybe it will still needs only one weight, that's something we have to check I guess

Reply:
The more information the better. If you have weights that you do not use: so be it.
But I don't believe so. In your case you should have nv+nv weights for x which means you have the same amount of weights than in the message. It might happen that a lot of weights are identical. But again this way you give more flexibility to the user with a small cost.

Reply:
The ocp is responsible to recreate the weights matrix. And it knows everything: x, robot_model, costs etc.

Reply:
are you talking about the activation weights when creating the residuals ? in this case I agree there are nv+nv weights or maybe something else, but for the cost itself it's always a scalar right ?

Reply:
Well you can see the quadratic cost like so:
$$
alpha * X^T W X
$$
What the weights inside the message will give you is W as a diagonal matrix.
Should we consider adding the alpha as well?
I would believe that W is enough to weight the cost. If all diagonal members are equal you can actually factorize and get $$ alpha X^T X $$.
Maybe I am mistaken...

Reply:
Yeah, I agree with you, Max; we can control alpha by controlling W.
And for me, having W_q and W_v and stacking them on the OCP side makes more sense than having W_{state} (since it is consistent with TrajectoryPoint definition)